### PR TITLE
Highlight current scope

### PR DESCRIPTION
**Pull request by theHamsta**
Highlights current scope. Should this be a built-in module or should I publish it as separate plugin? Then, I would just try to merge the first commit of this PR that adds a `highlight_node` utility function.

### CODE REVIEW

**Comment by theHamsta** [comment]
Would be probably more useful to highlight the scope at the sign-column.

**Comment by theHamsta** [comment]
highlight_scope

File: lua/nvim-treesitter/ts_utils.lua [hunk 1]
```diff
@@ -313,4 +313,17 @@ function M.find_usages(node, scope_node, bufnr)
   return usages
 end
 
+function M.highlight_node(node, buf, hl_namespace, hl_group)
+  if not node then return end
+  M.highlight_range({node:range()}, buf, hl_namespace, hl_group)
+end
+
+function M.highlight_range(range, buf, hl_namespace, hl_group)
```
Comment (theHamsta) [inline lua/nvim-treesitter/ts_utils.lua:321]: Should probably use *vim.highlight.range()*

**Comment by theHamsta** [comment]
It's also possible to do an inverse highlighting (dim everything but the current scope)

inverse_highlight

but this conflicts with treesitter highlighting.

**Comment by vigoux** [comment]
Ouch, if that's not a killer feature, I don't know what is ! Do you have some docs on how to have the current scope within the sign column too ?

**Comment by theHamsta** [comment]
Sign column is not implemented. I don't know how to do it without setting signs.

**Pull request merged**